File: CPP/Petshop.cpp
#include <iostream>
#include <string>

using namespace std;

class Petshop
{
private:
    int id;
    string nama_produk;
    int harga_produk;
    int stok_produk;

public:
    Petshop()
    {
    }
    Petshop(int id, string nama_produk, int harga_produk, int stok_produk)
    {
        this->id = id;
        this->nama_produk = nama_produk;
        this->harga_produk = harga_produk;
        this->stok_produk = stok_produk;
    }

    void set_id(int id)
    {
        this->id = id;
    }
    int get_id()
    {
        return this->id;
    }
    void set_nama_produk(string nama_produk)
    {
        this->nama_produk = nama_produk;
    }
    string get_nama_produk()
    {
        return this->nama_produk;
    }

    void set_harga_produk(int harga_produk)
    {
        this->harga_produk = harga_produk;
    }
    int get_harga_produk()
    {
        return this->harga_produk;
    }

    void set_stok_produk(int stok_produk)
    {
        this->stok_produk = stok_produk;
    }
    int get_stok_produk()
    {
        return this->stok_produk;
    }

    ~Petshop()
    {
        // cout << "Destructor called" << endl;
    }
};

File: PYTHON/Petshop.py
import string

class Petshop:
    def __init__(self, id="", nama_produk="", harga_produk=0, stok_produk=0):
        self.id = id
        self.nama_produk = nama_produk
        self.harga_produk = harga_produk
        self.stok_produk = stok_produk

    def __str__(self):
        return self.id + " " + self.nama_produk + " " + str(self.harga_produk) + " " + str(self.stok_produk)
    
    #getter and setter
    def getId(self):
        return self.id
    def setId(self, id):
        self.id = id
    def getNamaProduk(self):
        return self.nama_produk
    def setNamaProduk(self, nama_produk):
        self.nama_produk = nama_produk
    def getHargaProduk(self):
        return self.harga_produk
    def setHargaProduk(self, harga_produk):
        self.harga_produk = harga_produk
    def getStokProduk(self):
        return self.stok_produk
    def setStokProduk(self, stok_produk):
        self.stok_produk = stok_produk
    

    #destructor
    def __del__(self):
        pass